sphinx/environment/__init__.py

```diff
@@ -296,6 +297,27 @@ def _config_status(*, old_config: Config | None, new_config: Config) -> tuple[in
                 extension = f'{len(extensions)}'
             return CONFIG_EXTENSIONS_CHANGED, f' ({extension!r})'
 
+        # Log any changes in configuration keys
+        if changed_keys := _differing_config_keys(old_config, new_config):
+            changed_num = len(changed_keys)
+            if changed_num == 1:
+                logger.info(
+                    __('1 configuration option has changed: %s'),
+                    changed_keys[0],
+                )
+            elif changed_num <= 10:
+                logger.info(
+                    __('%d configuration options have changed: %s'),
+                    changed_num,
+                    ', '.join(changed_keys),
+                )
+            else:
+                logger.info(
+                    __('%d configuration options have changed: %s, ...'),
+                    changed_num,
+                    ', '.join(changed_keys[:10]),
+                )
```

Comment:
Very happy to have better wording suggestions here...

Reply:
How about having the entire list of changes in verbose mode? For the info log, I would say (use -v to show all items that were changed) or something like that (sorry it's 2 AM and I won't be there for the next few days...)